### PR TITLE
Initial support for Data Migration Service

### PR DESCRIPTION
**Pull request by vchske**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by promptws** [comment]
View a preview at https://prompt.ws/r/Azure/azure-cli/6159
This is an **experimental preview** for @microsoft users.

**Comment by derekbekoe** [comment]
Please rebase with the latest on dev branch. There was a fix just merged to fix a Travis CI change.

**Comment by derekbekoe** [comment]
Adding @sptramer to review the help text.

File: src/command_modules/azure-cli-dms/azure/cli/command_modules/dms/custom.py [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+
+    return client.create_or_update(group_name=resource_group_name,
+                                   service_name=service_name,
+                                   project_name=project_name,
+                                   task_name=task_name,
+                                   properties=migration_properties)
+
+
+def list_tasks(client, resource_group_name, service_name, project_name, task_type=None):
+    return list(client.tasks.list(group_name=resource_group_name,
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-dms/azure/cli/command_modules/dms/custom.py]: You don't need to wrap your SDK call in `list()`.
The CLI handles that.

**Pull request merged**